test/external_tests.py

```diff
@@ -74,9 +74,10 @@ def run_external_tests(args: dict):
                 solc_binary_path,
                 {k: all_test_scripts[k] for k in selected_tests},
             )
-    raise ExternalTestNotFound(
-        "External test was not selected. Please use --run or --run-all option"
-    )
+        else:
+            raise ExternalTestNotFound(
+                "External test was not selected. Please use --run or --run-all option"
+            )
```

Comment:
It's fine that way, and I didn't want to be *too* picky about this in the original PR (I already made a lot of suggestions and this one is not an absolute rule or anything) but if think we should prefer a flatter structure with early exits. So if we're changing it anyway, I'd do it like this:

```python
    if args["run_all"]:
        run_test_scripts(solc_binary_type, solc_binary_path, all_test_scripts)
        return

    if args["selected_tests"]:
        # raise ...

    unrecognized_tests = set(selected_tests) - set(all_test_scripts.keys())
    if unrecognized_tests != set():
        # raise

    # do the default thing
```

I.e. deal with special cases first (which is usually short) so that you're not so deeply nested when you get to the main case.

Reply:
Nice. No worries, it indeed provide much better readability. But the assertion in that case will always fail, since `selected_tests` has a default of a empty list, so I will assert if it is empty in that case. I will push the changes in a minute.

Reply:
Yeah, wasn't sure about it but it's an assert so if it's wrong, you'd see that when you run it and correct it anyway :)